resurfemg/helper_functions.py

```diff
@@ -1204,45 +1273,46 @@ def area_under_curve(
     :type end_curve: float
     :param smooth_algorithm: algorithm for smoothing
     :type smooth_algorithm: str
-
     :returns: area; area under the curve
     :rtype: float
     """
+    if not (0 <= end_curve <= 100):
+        raise ValueError(
+            'end_curve must be between 0 and 100, '
+            'but {} given'.format(end_curve),
+        )
+    if smooth_algorithm not in ('none', 'mid_savgol'):
+        raise ValueError(
+            'Possible values for smooth_algorithm are none and mid_savgol, '
+            'but {} given'.format(smooth_algorithm),
+        )
 
-    new_array = array[start_index: (end_index+1)]
-    max_ind = (new_array.argmax())
-    if end_curve > 100 or end_curve < 0:
-        print('You picked an impossible number for end_curve')
-        area = 'mistake'
-    else:
-        end_curve = end_curve/100
-        if array[start_index] < array[end_index]:
-            print('You picked an end point above baseline,')
-            print('caution with end_curve variable!')
-        if smooth_algorithm == 'none':
-            absolute_val_array = np.abs(
-                new_array[max_ind:] - new_array.max() * end_curve)
-            smallest_difference_index = absolute_val_array.argmin()
-            smallest_difference_index = smallest_difference_index + max_ind
-            area = np.sum(new_array[:smallest_difference_index])
-        if smooth_algorithm == 'mid_savgol':
-            new_array = savgol_filter(
-                new_array,
-                int(len(new_array)),
-                2,
-                deriv=0,
-                delta=1.0,
-                axis=- 1,
-                mode='interp',
-                cval=0.0,
-            )
-            absolute_val_array = np.abs(
-                new_array[max_ind:] - new_array.max() * end_curve)
-            smallest_difference_index = absolute_val_array.argmin()
-            smallest_difference_index = smallest_difference_index + max_ind
-            area = np.sum(new_array[:smallest_difference_index])
-
-    return area
+    if array[start_index] < array[end_index]:
+        logging.warning(
+            'You picked an end point above baseline, '
+            'caution with end_curve variable!',
+        )
```

Comment:
I agree it might be better to have an average we look at, but it's mathematically tricky. If the baseline moves from 1 to 2 and we have a peak of 1000 it's fine, but if we had a peak of 4 we are in trouble. This needs another issue and refactor, but for this release, the warning should be enought

Reply:
So i'll push a new version